lib/qesapdeployment.pm

```diff
@@ -745,4 +749,112 @@ sub qesap_get_az_resource_group {
     return $result;
 }
 
+=head3 qesap_calculate_az_address_range
+
+Calculate the vnet and subnet address
+ranges. The format is 10.ip2.ip3.0/21 and
+ /24 respectively. ip2 and ip3 are calculated
+ using the slot number as seed.
+
+=over 1
+
+=item B<SLOT> - integer to be used as seed in calculating addresses
+
+=back
+
+=cut
+
+sub qesap_calculate_az_address_range {
+    my %args = @_;
+    croak 'Missing mandatory slot argument' unless $args{slot};
+
+    die "Invalid 'slot' argument - valid values are 1-8192" if ($args{slot} > 8192 || $args{slot} < 1);
+
+    my $offset = ($args{slot} - 1) * 8;
+
+    # addresses are of the form 10.ip2.ip3.0/21 and /24 respectively
+    #ip2 gets incremented when is >=256
+    my $ip2 = int($offset / 256);
```

Comment:
No need to convert this into an integer, when you're passing it below in L781 & L782 to `sprintf()` using `%d`  ... `%d` is an integer casting and will do the conversion:
```
# perl
my $t = 0.9999;
print sprintf('string created with sprintf. [%d] was a float', $t), "\n";
string created with sprintf. [0] was a float
```

Reply:
Even nicer catch!

Reply:
It *may* be good for visibility to someone trying to understand the code, though